vm/src/stdlib/time.rs

```diff
@@ -378,22 +378,19 @@ mod time {
     impl TryFromObject for PyStructTime {
         fn try_from_object(vm: &VirtualMachine, seq: PyObjectRef) -> PyResult<Self> {
             let seq: Vec<_> = seq.try_to_value(vm)?;
-            if seq.len() != 9 {
-                return Err(
-                    vm.new_type_error("time.struct_time() takes a sequence of length 9".to_owned())
-                );
-            }
-            let mut i = seq.into_iter();
+            let v: [PyObjectRef; 9] = seq.try_into().map_err(|_| {
+                vm.new_type_error("time.struct_time() takes a sequence of length 9".to_owned())
+            })?;
```

Comment:
Will you check #3613 for this? The idea using array looks good. Maybe we can put it in the general way.